File: android/src/main/kotlin/com/luk/flutter/sdk/luk/util/ResultUtil.kt
package com.luk.flutter.sdk.luk.util


object ResultUtil {

    fun buildResult(
        code: Int,
        msg: String,
        data: HashMap<String, Any> = java.util.HashMap()
    ): HashMap<String, Any> {
        val result = HashMap<String, Any>()
        result["code"] = code
        result["msg"] = msg
        result["data"] = data
        return result
    }

}

File: lib/util/result_util.dart
import 'package:luk/bean/result_bean.dart';

class ResultUtil {
  ResultUtil._();

  static ResultBean getResult(dynamic args) {
    if (args is Map) {
      int code = args["code"] as int;
      String msg = args["msg"] as String;
      Map data = (args["data"] as Map?) ?? {};
      return ResultBean(code, msg, data);
    }
    return ResultBean(-1, "参数错误！", {});
  }
}
